From linux-usb-devel-admin@lists.sourceforge.net Sun Oct  9 18:34:51 2005
From: Ben Dooks <ben@fluff.org.uk>
To: linux-usb-devel@lists.sourceforge.net
Message-ID: <20051010013215.GA4522@home.fluff.org>
Content-Disposition: inline
Subject: USB: add owner initialisation to host drivers
Date: Mon, 10 Oct 2005 02:32:15 +0100

Add .owner initialisation to the device drivers
in drivers/usb/host so that when built as module
the device_driver refers to the owning module

Signed-off-by: Ben Dooks <ben-linux@fluff.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ehci-pci.c     |    1 +
 drivers/usb/host/ohci-au1xxx.c  |    1 +
 drivers/usb/host/ohci-lh7a404.c |    1 +
 drivers/usb/host/ohci-omap.c    |    1 +
 drivers/usb/host/ohci-pci.c     |    1 +
 drivers/usb/host/ohci-ppc-soc.c |    1 +
 drivers/usb/host/sl811-hcd.c    |    1 +
 drivers/usb/host/uhci-hcd.c     |    1 +
 8 files changed, 8 insertions(+)

--- gregkh-2.6.orig/drivers/usb/host/ohci-au1xxx.c
+++ gregkh-2.6/drivers/usb/host/ohci-au1xxx.c
@@ -264,6 +264,7 @@ static int ohci_hcd_au1xxx_drv_resume(st
 
 static struct device_driver ohci_hcd_au1xxx_driver = {
 	.name		= "au1xxx-ohci",
+	.owner		= THIS_MODULE,
 	.bus		= &platform_bus_type,
 	.probe		= ohci_hcd_au1xxx_drv_probe,
 	.remove		= ohci_hcd_au1xxx_drv_remove,
--- gregkh-2.6.orig/drivers/usb/host/ohci-lh7a404.c
+++ gregkh-2.6/drivers/usb/host/ohci-lh7a404.c
@@ -244,6 +244,7 @@ static int ohci_hcd_lh7a404_drv_resume(s
 
 static struct device_driver ohci_hcd_lh7a404_driver = {
 	.name		= "lh7a404-ohci",
+	.owner		= THIS_MODULE,
 	.bus		= &platform_bus_type,
 	.probe		= ohci_hcd_lh7a404_drv_probe,
 	.remove		= ohci_hcd_lh7a404_drv_remove,
--- gregkh-2.6.orig/drivers/usb/host/ohci-omap.c
+++ gregkh-2.6/drivers/usb/host/ohci-omap.c
@@ -492,6 +492,7 @@ static int ohci_omap_resume(struct devic
  */
 static struct device_driver ohci_hcd_omap_driver = {
 	.name		= "ohci",
+	.owner		= THIS_MODULE,
 	.bus		= &platform_bus_type,
 	.probe		= ohci_hcd_omap_drv_probe,
 	.remove		= ohci_hcd_omap_drv_remove,
--- gregkh-2.6.orig/drivers/usb/host/ohci-pci.c
+++ gregkh-2.6/drivers/usb/host/ohci-pci.c
@@ -216,6 +216,7 @@ MODULE_DEVICE_TABLE (pci, pci_ids);
 static struct pci_driver ohci_pci_driver = {
 	.name =		(char *) hcd_name,
 	.id_table =	pci_ids,
+	.owner =	THIS_MODULE,
 
 	.probe =	usb_hcd_pci_probe,
 	.remove =	usb_hcd_pci_remove,
--- gregkh-2.6.orig/drivers/usb/host/ohci-ppc-soc.c
+++ gregkh-2.6/drivers/usb/host/ohci-ppc-soc.c
@@ -193,6 +193,7 @@ static int ohci_hcd_ppc_soc_drv_remove(s
 
 static struct device_driver ohci_hcd_ppc_soc_driver = {
 	.name		= "ppc-soc-ohci",
+	.owner		= THIS_MODULE,
 	.bus		= &platform_bus_type,
 	.probe		= ohci_hcd_ppc_soc_drv_probe,
 	.remove		= ohci_hcd_ppc_soc_drv_remove,
--- gregkh-2.6.orig/drivers/usb/host/sl811-hcd.c
+++ gregkh-2.6/drivers/usb/host/sl811-hcd.c
@@ -1831,6 +1831,7 @@ sl811h_resume(struct device *dev)
 struct device_driver sl811h_driver = {
 	.name =		(char *) hcd_name,
 	.bus =		&platform_bus_type,
+	.owner =	THIS_MODULE,
 
 	.probe =	sl811h_probe,
 	.remove =	__devexit_p(sl811h_remove),
--- gregkh-2.6.orig/drivers/usb/host/uhci-hcd.c
+++ gregkh-2.6/drivers/usb/host/uhci-hcd.c
@@ -831,6 +831,7 @@ MODULE_DEVICE_TABLE(pci, uhci_pci_ids);
 static struct pci_driver uhci_pci_driver = {
 	.name =		(char *)hcd_name,
 	.id_table =	uhci_pci_ids,
+	.owner =	THIS_MODULE,
 
 	.probe =	usb_hcd_pci_probe,
 	.remove =	usb_hcd_pci_remove,
--- gregkh-2.6.orig/drivers/usb/host/ehci-pci.c
+++ gregkh-2.6/drivers/usb/host/ehci-pci.c
@@ -370,6 +370,7 @@ MODULE_DEVICE_TABLE(pci, pci_ids);
 static struct pci_driver ehci_pci_driver = {
 	.name =		(char *) hcd_name,
 	.id_table =	pci_ids,
+	.owner = 	THIS_MODULE,
 
 	.probe =	usb_hcd_pci_probe,
 	.remove =	usb_hcd_pci_remove,
